src/DotNet/ModuleContext.cs

```diff
@@ -9,6 +11,7 @@ namespace dnlib.DotNet {
 	public class ModuleContext {
 		IAssemblyResolver assemblyResolver;
 		IResolver resolver;
+		Dictionary<byte, OpCode[]> experimentalOpCodes;
```

Comment:
I think an `OpCode[][]` is better. We know the exact size of it and it's small. The lookups will be very fast as well.

Lazy initialization shouldn't be needed, it's small so you can allocate in the ctor and set the field to `readonly`.

The `ExperimentalOpCodes` property would then need to be removed and replaced with a method that returns the opcode (`TryGetExperimentalOpCode()`)

Reply:
So just to be clear, are you suggesting an array of size 12 (to which we map the range `0xf0`..`0xfb`) where each element is an array of 256 `OpCode` references?

Reply:
Yes an array of size 12 where each element is null. Initialization of this array should be done by the app before module is opened so Interlocked/etc isn't needed.